### DISCUSSION TITLE
Adding free space calculation

### DISCUSSION THREAD

**Discussion by Lagor31** [spaskalev/buddy_alloc#48] [General]
Hi, 
would it be possible to implement a way to tell the remaining space in the buddy?
I understand that, for instace, if you have 1Mb free does not necessarily mean that it's contiguous but, since i'm using it in my kernel, I'd love to see whether the memory system is performing as expected by checking the amount of memory that it's allocating/freeing.

Thanks, cheers.
FG

**Comment by spaskalev** [comment]
Certainly :) I'll implement that and ping you in PR

**Reply by Lagor31** [reply]
TY!

**Reply by spaskalev** [reply]
Please check https://github.com/spaskalev/buddy_alloc/pull/49 Cheers!

**Reply by Lagor31** [reply]
Hi, sorry for the late reply, life has gotten in the way of this :p  
I'll check your feature and I'll let you know asap.
In the meantime, thanks!

Cheers.

**Reply by Lagor31** [reply]
Thank you, it works as expected!
Keep up the good work.
Cheers!

**Comment by spaskalev** [comment]
Confirmed to work, resolving :)
